applications/CompressiblePotentialFlowApplication/custom_elements/adjoint_base_potential_flow_element.cpp

```diff
@@ -13,108 +13,85 @@
     {
         mpPrimalElement->CalculateLocalSystem(rLeftHandSideMatrix,
                                               rRightHandSideVector,
                                               rCurrentProcessInfo);
     }
 
     template <class TPrimalElement>
-    void AdjointPotentialFlowElement<TPrimalElement>::CalculateLeftHandSide(MatrixType& rLeftHandSideMatrix,
-                                       ProcessInfo& rCurrentProcessInfo) 
+    void AdjointBasePotentialFlowElement<TPrimalElement>::CalculateLeftHandSide(MatrixType& rLeftHandSideMatrix,
+                                       ProcessInfo& rCurrentProcessInfo)
     {
         MatrixType tmp;
         mpPrimalElement->CalculateLeftHandSide(tmp, rCurrentProcessInfo);
-        rLeftHandSideMatrix = trans(tmp);                                    
+        rLeftHandSideMatrix = trans(tmp);
     }
 
     template <class TPrimalElement>
-    void AdjointPotentialFlowElement<TPrimalElement>::CalculateRightHandSide(VectorType& rRightHandSideVector,
-                                        ProcessInfo& rCurrentProcessInfo) 
+    void AdjointBasePotentialFlowElement<TPrimalElement>::CalculateRightHandSide(VectorType& rRightHandSideVector,
+                                        ProcessInfo& rCurrentProcessInfo)
     {
         mpPrimalElement->CalculateRightHandSide(rRightHandSideVector,
                                                 rCurrentProcessInfo);
     }
 
     template <class TPrimalElement>
-    void AdjointPotentialFlowElement<TPrimalElement>::GetValueOnIntegrationPoints(const Variable<double>& rVariable,
+    void AdjointBasePotentialFlowElement<TPrimalElement>::GetValueOnIntegrationPoints(const Variable<double>& rVariable,
             std::vector<double>& rValues,
-            const ProcessInfo& rCurrentProcessInfo) 
+            const ProcessInfo& rCurrentProcessInfo)
     {
         mpPrimalElement->GetValueOnIntegrationPoints(rVariable,rValues,rCurrentProcessInfo);
     }
-    
+
     template <class TPrimalElement>
-    void AdjointPotentialFlowElement<TPrimalElement>::GetValueOnIntegrationPoints(const Variable<array_1d<double,3> >& rVariable,
+    void AdjointBasePotentialFlowElement<TPrimalElement>::GetValueOnIntegrationPoints(const Variable<array_1d<double,3> >& rVariable,
             std::vector< array_1d<double,3> >& rValues,
-            const ProcessInfo& rCurrentProcessInfo) 
+            const ProcessInfo& rCurrentProcessInfo)
     {
         mpPrimalElement->GetValueOnIntegrationPoints(rVariable,rValues,rCurrentProcessInfo);
     }
 
     template <class TPrimalElement>
-    void AdjointPotentialFlowElement<TPrimalElement>::GetValuesVector(Vector& rValues, int Step) 
+    void AdjointBasePotentialFlowElement<TPrimalElement>::GetValuesVector(Vector& rValues, int Step)
     {
         KRATOS_TRY
-        const AdjointPotentialFlowElement& r_this = *this;
+        const AdjointBasePotentialFlowElement& r_this = *this;
```

Comment:
Why do you need to make it `const`? To have a threadsafe access?

Reply:
Yes! This was implemented by @inigolcanalejo  in the incompressible element follwoing suggestions by @philbucher (if im not mistaken). @philbucher do you have more insight on this?

Reply:
The point is that you are accessing to the element database. Provided that you've initialized the variables you are accessing to, there shouldn't be any problem. @philbucher am I wrong?

Reply:
@rubenzorrilla you are absolutely right
=> accessing a variable of the element will [allocate this variable if it does not exist yet](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/containers/data_value_container.h#L223) (=> this is the correct behavior of the `DataValueContainer`), which might not always be desired. This ofc cannot happen with a const version of the element

I had in mind to clearly tell others that take a look at this code: Here I do NOT want to allocate anything.

Ofc it also works if I correctly allocate the memory beforehand, but this is done outside of the elements. Now someone writes a new solver, then the behavior changes unknowingly.

(BTW not so long ago I had to debug sth for quite some time exactly bcs of what I described :) )

Reply:
Long story short, in here is that you don't want to allocate the memory and initialize the value in the `Initialize()` of the element but to assume that it has been done beforehand by any other utility. Am I wrong?

May I ask why you don't want this allocation to be done? (just for me to know).

Reply:
I encountered some execution errors in the current when running in FullDebug when using non historical variables. I have "fixed" them by using the same technique as here, but im still not clear if this is the intended way to do so. 

Since this discussion goes a bit further than the scope of this PR I suggest following the discussion in #5189 , where I  pushed the hotfix.

Reply:
> May I ask why you don't want this allocation to be done? (just for me to know).

I am not saying that I do not want the allocation, but I want to point out the behavior since it can affect how you write the code.
It really depends on the usecase